inst/qview/qview.R

```diff
@@ -0,0 +1,194 @@
+  queue_obj = rrsq::RSimpleQueue$new()
+
+  # Initialize our data frame, "queue_data".
+  # We pull the source data with queue_obj$get_jobs, then convert it to a data frame and
+  #   sort the contents by ID (descending). We then assign this to as a component of a
+  #   reactiveValue for easy-access in other reactive contexts.
+  .rv <- reactiveValues()
+  .rv$queue_data <-
+    rrsq::jobs_to_df(queue_obj$get_jobs()) %>%
+    dplyr::filter(user == .user) %>%
+    dplyr::arrange(desc(id))
+
+  .return_values <- reactiveValues()
+
+  # Create an observe event that re-runs (via invalidation) every 1000 milliseconds.
+  # In this section, we grab an updated version of our data frame and compare it
+  #   to the one currently in use. If any differences are found, we upate the data
+  #   frame in use. This prevents the UI from constantly refreshing every time it
+  #   checks for new data.
+  observe({
+    invalidateLater(.refresh_interval)
+    new_data <-
+      rrsq::jobs_to_df(queue_obj$get_jobs()) %>%
+      dplyr::filter(user == .user) %>%
+      dplyr::arrange(desc(id))
+
+    if (
+      !isTRUE(all_equal(.rv$queue_data, new_data))
+      ) {
+      .rv$queue_data <- new_data
+    }
+  })
+
+  # Create the DataTable (DT) object to be rendered.
+  #   Using the data in our reactiveValue (.rv$queue_data), we render a Data Table object.
+  #   While doing so, we add a column of buttons to the Data Table with the help of the
+  #   function "cancel_button_creator". "cancel_button_creator" needs our namespacing
+  #   function in order to properly namespace the buttons with this module's UI namespace.
+  output$queue_table  <- DT::renderDT(
+    {
+
+      if(nrow(.rv$queue_data) > 0) {
+        cancel_button_column <- map_chr(
+          1:nrow(.rv$queue_data),
+          cancel_button_creator,
+          .ns = ns
+        )
+      } else {
+        cancel_button_column <- list()
+      }
+
+      .rv$queue_data %>%
+        mutate(cancel_button = cancel_button_column) %>%
+        dplyr::select(cancel_button, everything())
+    },
+    escape = FALSE,
+    selection = "single"
+  )
+
+
+
+  # Create an eventReactive for when Cancel buttons are pushed.
+  # This eventReactive triggers whenever a Cancel button is pushed. All Cancel buttons are
+  #   treated as as if they had the id ns("cancel_button") (this is set up in
+  #   cancel_button_creator). It saves the index of the row where the button was clicked.
+  cancel_index <- eventReactive(input$cancel_button, {
+    return(input$cancel_button)
+  })
+
+  # Submit the cancellation request
+  # Whenever the cancel_index changes (after a cancel button is clicked), we get use the
+  #   index to grab the relevant row from the table, from which we grab the job ID to be
+  #   cancelled. We then use the queue_obj to submit the cancellation request.
+  observe({
+    index <- as.numeric(cancel_index())
+    selected_row <- .rv$queue_data %>% slice(index)
+    queue_obj$cancel_job(selected_row$id)
```

Comment:
does cancel job return anything? I thought? I returned a 200 if the job was cancelled successfully. If it does, you might want to check just to make sure

Reply:
after a job is cancelled you might also want to trigger getting the current state of the jobs (eg with the newly cancelled job) reflect in the status, so in case there is a longer timeout period the data doesn't get stale between cancelling it, and getting the UI refreshed.

Reply:
#10 added.